app/models/job_execution.rb

```diff
@@ -37,7 +38,12 @@ def wait!
   end
 
   def stop!
-    @executor.stop!
+    @executor.stop! 'INT'
+    self.class.stop_timeout.times do
```

Comment:
if `stop_timeout` is a time (like `15.seconds`), you shouldn't be doing `stop_timeout.times`